src/HelicityAngles.cxx

```diff
@@ -9,109 +9,70 @@
 #include "ParticleCombination.h"
 #include "StatusManager.h"
 
-namespace yap {
+#include "logging.h"
 
-//-------------------------
-HelicityAngles::HelicityAngles(Model& m) :
-    StaticDataAccessor(m, equal_up_and_down),
-    Phi_(RealCachedValue::create(*this)),
-    Theta_(RealCachedValue::create(*this))
-{
-    registerWithModel();
-}
+#include <assert.h>
 
-//-------------------------
-void HelicityAngles::addToStaticDataAccessors()
-{
-    // look for Model's FourMomenta_
-    auto it_fm = std::find(staticDataAccessors().begin(), staticDataAccessors().end(), model()->fourMomenta().get());
-    if (it_fm == staticDataAccessors().end())
-        throw exceptions::Exception("HelicityAngles cannot be registered with the model before FourMomenta", "HelicityAngles::registerWithModel");
-    // add this to just after FourMomenta_
-    const_cast<StaticDataAccessorVector&>(staticDataAccessors()).insert(it_fm + 1, this);
-}
+namespace yap {
 
-//-------------------------
-double HelicityAngles::phi(const DataPoint& d, const std::shared_ptr<const ParticleCombination>& pc) const
-{
-    return Phi_->value(d, symmetrizationIndex(pc));
-}
 
 //-------------------------
-double HelicityAngles::theta(const DataPoint& d, const std::shared_ptr<const ParticleCombination>& pc) const
+const std::array<double, 2>& HelicityAngles::helicityAngles(const DataPoint& d, const StatusManager& sm, const std::shared_ptr<const ParticleCombination>& pc) const
 {
-    return Theta_->value(d, symmetrizationIndex(pc));
-}
+    // check if DataPoint is currently in cache
+    if (cachedForDataPoint_[&sm] == &d) {
+        // find entry
+        for (const auto& kv : cachedAngles_[&sm])
```

Comment:
Avoid using `kv`. Replace it with what the actual key and value are: `pc_angles`.